### PR TITLE
EffMessage - add support for optional sender

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to add support for a sender in the message effect. 
Minecraft added a player ignore system in 1.16.4. Using the Player#sendMessage method in Bukkit, sends the message as a system message, thus not allowing players to ignore messages. This update will allow for that.

I also updated the Paper dependency to 1.16.4 since the method including UUID (of sender) was recently added.

---
**Target Minecraft Versions:** 1.16.4+
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/effects/EffMessage.java [hunk 1]
```diff
@@ -37,20 +40,34 @@
...
 import ch.njol.util.Kleenean;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.TextComponent;
 
 @Name("Message")
 @Description({"Sends a message to the given player. Only styles written",
-		"in given string or in <a href=expressions.html#ExprColoured>formatted expressions</a> will be parsed."})
+		"in given string or in <a href=expressions.html#ExprColoured>formatted expressions</a> will be parsed.",
+		"Adding an optional sender allows the messages to be sent as if a specific player sent them.",
+		"This is useful with Minecraft 1.16.4's new chat ignore system."})
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffMessage.java:50]: The chat blocking system doesn't work if you don't send it as a player, right? If so, maybe it'd be better to say something like: `This allows the chat ignore system added in Minecraft 1.16.4 to block these messages`

Up for discussion :D

File: src/main/java/ch/njol/skript/effects/EffMessage.java [hunk 2]
```diff
@@ -37,20 +40,34 @@
...
+		"\tsend \"[%player%] >> %message%\" to all players from player"})
+@RequiredPlugins("Minecraft 1.16.4+ for optional sender")
+@Since("1.0, 2.2-dev26 (advanced features), INSERT VERSION (optional sender)")
 public class EffMessage extends Effect {
 	
+	private static final boolean SUPPORTS_SENDER;
+	
 	static {
-		Skript.registerEffect(EffMessage.class, "(message|send [message[s]]) %strings% [to %commandsenders%]");
+		SUPPORTS_SENDER = Skript.classExists("org.bukkit.command.CommandSender$Spigot") && Skript.methodExists(CommandSender.Spigot.class, "sendMessage", UUID.class, BaseComponent.class);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffMessage.java:66]: I don't know if this really matters, but would it be better to just initialize the variable above?

File: src/main/java/ch/njol/skript/effects/EffMessage.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/effects/EffMessage.java:50]: Yes, Good call. Ill all a little more to the description.

File: src/main/java/ch/njol/skript/effects/EffMessage.java (hunk 2 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/effects/EffMessage.java:66]: Yeah I could do that.

File: src/main/java/ch/njol/skript/effects/EffMessage.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/effects/EffMessage.java]: done 👍

File: src/main/java/ch/njol/skript/effects/EffMessage.java (hunk 2 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/effects/EffMessage.java]: done 👍

**Review by APickledWalrus — DISMISSED** [review]

**Pull request merged**